src/ChunkDef.h

```diff
@@ -85,6 +85,50 @@ class cChunkDef
 	/** The storage wrapper used for compressed nibbledata residing in RAMz */
 	typedef std::vector<NIBBLETYPE> COMPRESSED_NIBBLETYPE;
 
+    static bool IsMoreThanChunkHeight(int v)
+    {
+        return v >= Height;
+    }
+
+    static bool IsLessThanChunkHeight(int v)
+    {
+        return v < 0;
+    }
+
+    static bool IsValidHeight(int v)
+    {
+        return !IsMoreThanChunkHeight(v) && !IsLessThanChunkHeight(v);
+    }
+
+    static bool IsMoreThanChunkWidth(int v)
+    {
+        return v >= Width;
+    }
+
+    static bool IsLessThanChunkWidth(int v)
+    {
+        return v < 0;
+    }
+
+    static bool IsValidWidth(int v)
+    {
+        return !IsMoreThanChunkWidth(v) && !IsLessThanChunkWidth(v);
+    }
+
+    static bool IsValidCoordinates(int x, int y, int z)
+    {
+        return IsValidWidth(x) && IsValidHeight(y) && IsValidWidth(z);
+    }
+
+    static void AssertCoordinatesIsValid(int x, int y, int z)
```

Comment:
No problem, I will make those change. I had trouble finding a variable name for the width methods. That's why I simply used v. The Width methods can be called with an x or a z. I didn't want to use x as a parameter as I tought user of the code would look for the method with a z. Do you have a suggestion for the parameter name?